# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/trunk/webrtc/modules/desktop_capture/x11/x_server_pixel_buffer.h
# Commit: 3f0981f37c1c
# Full Hash: 3f0981f37c1c499ad52016e526239b390703ba2a
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2019-04-17 21:47:29
# Regressor Bug: 1496359
# File Overlap Count: 1
# Description:
#   Bug 1496359 - Add PipeWire support to desktop capture; r=ng
#   
#   This is an import of upstream commit 318da51f99f91e3de1192f29d7f1824958f9f13e.
#   
#   Depends on D27367
# ==============================================================================

diff -r ceed4e0b3373 -r 3f0981f37c1c media/webrtc/trunk/webrtc/modules/desktop_capture/x11/x_server_pixel_buffer.h
--- a/media/webrtc/trunk/webrtc/modules/desktop_capture/x11/x_server_pixel_buffer.h	Mon Apr 15 17:12:31 2019 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,84 +0,0 @@
-/*
- *  Copyright (c) 2013 The WebRTC project authors. All Rights Reserved.
- *
- *  Use of this source code is governed by a BSD-style license
- *  that can be found in the LICENSE file in the root of the source
- *  tree. An additional intellectual property rights grant can be found
- *  in the file PATENTS.  All contributing project authors may
- *  be found in the AUTHORS file in the root of the source tree.
- */
-
-// Don't include this file in any .h files because it pulls in some X headers.
-
-#ifndef MODULES_DESKTOP_CAPTURE_X11_X_SERVER_PIXEL_BUFFER_H_
-#define MODULES_DESKTOP_CAPTURE_X11_X_SERVER_PIXEL_BUFFER_H_
-
-#include "modules/desktop_capture/desktop_geometry.h"
-#include "rtc_base/constructormagic.h"
-
-#include <X11/Xutil.h>
-#include <X11/extensions/XShm.h>
-
-namespace webrtc {
-
-class DesktopFrame;
-
-// A class to allow the X server's pixel buffer to be accessed as efficiently
-// as possible.
-class XServerPixelBuffer {
- public:
-  XServerPixelBuffer();
-  ~XServerPixelBuffer();
-
-  void Release();
-
-  // Allocate (or reallocate) the pixel buffer for |window|. Returns false in
-  // case of an error (e.g. window doesn't exist).
-  bool Init(Display* display, Window window);
-
-  bool is_initialized() { return window_ != 0; }
-
-  // Returns the size of the window the buffer was initialized for.
-  DesktopSize window_size() { return window_rect_.size(); }
-
-  // Returns the rectangle of the window the buffer was initialized for.
-  const DesktopRect& window_rect() { return window_rect_; }
-
-  // Returns true if the window can be found.
-  bool IsWindowValid() const;
-
-  // If shared memory is being used without pixmaps, synchronize this pixel
-  // buffer with the root window contents (otherwise, this is a no-op).
-  // This is to avoid doing a full-screen capture for each individual
-  // rectangle in the capture list, when it only needs to be done once at the
-  // beginning.
-  void Synchronize();
-
-  // Capture the specified rectangle and stores it in the |frame|. In the case
-  // where the full-screen data is captured by Synchronize(), this simply
-  // returns the pointer without doing any more work. The caller must ensure
-  // that |rect| is not larger than window_size().
-  bool CaptureRect(const DesktopRect& rect, DesktopFrame* frame);
-
- private:
-  void ReleaseSharedMemorySegment();
-
-  void InitShm(const XWindowAttributes& attributes);
-  bool InitPixmaps(int depth);
-
-  Display* display_ = nullptr;
-  Window window_ = 0;
-  DesktopRect window_rect_;
-  XImage* x_image_ = nullptr;
-  XShmSegmentInfo* shm_segment_info_ = nullptr;
-  XImage* x_shm_image_ = nullptr;
-  Pixmap shm_pixmap_ = 0;
-  GC shm_gc_ = nullptr;
-  bool xshm_get_image_succeeded_ = false;
-
-  RTC_DISALLOW_COPY_AND_ASSIGN(XServerPixelBuffer);
-};
-
-}  // namespace webrtc
-
-#endif  // MODULES_DESKTOP_CAPTURE_X11_X_SERVER_PIXEL_BUFFER_H_